### DISCUSSION TITLE
Stutter Automation

### DISCUSSION THREAD

**Discussion by nilscreates** [SynthstromAudible/DelugeFirmware#853] [Ideas]
Is there a particular reason why Stutter can be automated? It's my most favourite knob on the deluge and I use it a lot to add organic and glitchy variation to beats and samples, and it would be fab to be able to record/automate those.

On that note, I did wonder why some other features (like e.g. saturation) can't be automated or mapped at all. Is this a technical constraint?

_Sorry, i've been banging on about this... would love to have the skills to contribute myself, but am too much of a noob and can only support through patreon._

**Comment by Rudement** [comment]
It’s currently on the nightly build.

**Reply by nilscreates** [reply]
Wahooo, thank you for letting me know! And apologies for not checking first

**Reply by sichtbeton** [reply]
well partially. you can automate it but the actual button press doesnt work. so there is no effect until you actually press the button.

**Comment by thejesiah** [comment]
Just realized this was in the nightly!
But I can't for the life of me figure out how to automate it being on/off? 
Love seeing those orange LEDs just go all over!
Thanks, Devs!

**Comment by worker-gray** [comment]
I heard you still have to push the encoder to trigger the effect, which I think rather defeats the purpose of automation. The default level should be zero and any entry for automation should register as 'on' at the respective time division, and then revert to zero where no automation has been entered.

**Reply by seangoodvibes** [reply]
It was never called stutter automation but rather stutter rate automation. That's all that is possible atm.

**Reply by worker-gray** [reply]
i think that should be remedied by full automation.

**Comment by seangoodvibes** [comment]
> i think that should be remedied by full automation.

That requires an entirely new sequencer to be built. It's not a small undertaking.

The sequencer as is built around changing elements of sound (specifically, notes or parameter nodes). It's not built as a sequencer of discrete actions / triggers - such as triggering stutter, changing parameter types, etc.

That's not to say I don't want to do this. I do. Just trying to say it's not a fast "remedy" by any means.

**Reply by worker-gray** [reply]
I apologize if I miscommunicated.
By 'full automation' I mean engaging and disengaging the effect through the automation lane.
'Default to zero' means the effect is off completely- which would register as no lit pads on the note entry grid.
I guess it would be reading 'something' into the buffer at all times- for efficiency's sake the largest available buffer in terms of beat division would probably be optimal-- assuming that is anything that can be accessed via code.
Then when an automation key is entered (via note grid) it would automatically engage the effect for the beat division prescribed for that 'row'-- basically like the other automation lanes except this one is off by default.

**Reply by worker-gray** [reply]
@seangoodvibes  I THANK YOU if you are the person who has undertaken the task of writing this code. Automation of these parameters in any state is a blessing and a long-time request of mine--from way back in '17!
You're the one doing the work and I apologize if I'm way off base here. I am but a humble user.

**Reply by worker-gray** [reply]
Thinking about it further, I can foresee potential conflicts (assuming there's one audio buffer) if you automate more than one to stutter at different rates at the same time or if you trigger an arrangement-wide stutter over a single track stutter. I don't know how the CPU would handle that kind of overlap. It might sound cool but I don't want my machine to start smoking at such a young age.
